confgen/api_internal.go

```diff
@@ -954,10 +982,18 @@ func fetchVolumeInfo(confMap conf.ConfMap) (whoAmI string, localVolumeGroupMap v
 
 	for volumeGroupName, volumeGroup = range globalVolumeGroupMap {
 		if whoAmI == volumeGroup.PrimaryPeer {
-			localVolumeGroupMap[volumeGroupName] = volumeGroup
 			for volumeName, volume = range volumeGroup.VolumeMap {
 				localVolumeMap[volumeName] = volume
 			}
+
+			var shared bool
+			shared, err = IsVolumeGroupSharedSMB(confMap, volumeGroupName)
```

Comment:
I take it this discerning VGs for SMB vs non-SMB (and non-either) is because each SMB VG has it's own smb.conf (and VIP).

Is there a corresponding func to fetch the non-SMB VGs (presumably for populating the "connon" /etc/exports)?

Reply:
There is **IsVolumeGroupSharedNFS()** to tell you if the VG needs NFS.

Reply:
Ed is correct about the reason for checking whether a VG has any SMB shares.  If it does, then it needs per VIP smb.conf files.

Volumes that are shared via NFS go in /etc/exports, not volume groups.  that code walks the volume list and checks `len(volume.nfsClientList) > 0`.